### ISSUE TITLE
Rename `units` to `locations`

### ISSUE THREAD

**Issue by timtroendle** [calliope-project/euro-calliope#349]
The name `units` that is used heavily in the workflow stems from "administrative units". We may want to consider renaming them to `locations`.

These are the reasons:

1. With eHighways being built in, not all locations are administrative.
2. The abbreviation `units` for "administrative units" isn't clear and many people may not even know what it means.
3. Calliope uses `locations` and we would be consistent.

**Comment by timtroendle** [comment]
This is up for discussions. @calliope-project/euro-calliope-devs happy to hear your opinions.

**Comment by irm-codebase** [comment]
@timtroendle I believe the preferred naming in Calliope 0.7 is nodes. So perhaps we should use that?

See: https://calliope.readthedocs.io/en/latest/math/base/

**Comment by jnnr** [comment]
I agree that a consistent naming will help. However, 'node' is a graph theory term and doesn't really fit in the context geospatial data processing of euro-calliope, in my view. I propose either **regions** or **spatial_units**, which were both terms we used consistently in our sprint week spatial resolution team.

**Comment by sjpfenninger** [comment]
As of Calliope 0.7 [we do use `nodes` inside Calliope itself](https://calliope.readthedocs.io/en/latest/creating/nodes/), not `locations`. But perhaps using `locations` in the euro-calliope processing chain all the way up to the end, where the Calliope model is built (where `nodes` will then be mandatory), helps clearly separate the data processing from the model building.

**Comment by brynpickering** [comment]
`regions` has the problem of possibly being mixed up with the `regional` resolution of the model. `spatial_units` could work. There's also `zones` and `geometries`

**Comment by irm-codebase** [comment]
How about `shapes`? It's shorter than `geometries` / `spatial_units`, avoids confusion with other model functions, and is a bit clearer than `zones` (imo).

- national_shapes
- regional_shapes
- continental_shapes
- ehighway_shapes
- etc...
